swift/src/IceImpl/ObjectPrx.h

```diff
@@ -129,8 +129,9 @@ ICEIMPL_API @interface ICEObjectPrx : NSObject
 #ifdef __cplusplus
 
 @interface ICEObjectPrx()
-@property (nonatomic, readonly) std::shared_ptr<Ice::ObjectPrx> prx;
--(nullable instancetype) initWithCppObjectPrx:(std::shared_ptr<Ice::ObjectPrx>)prx;
+// prx is always set; we just need a field with a default constructor.
+@property (nonatomic, readonly) std::optional<Ice::ObjectPrx> prx;
+-(nonnull instancetype) initWithCppObjectPrx:(const Ice::ObjectPrx&)prx;
```

Comment:
In the previous version, we could init with a null proxy and get back nil ... which I find pretty unexpected.

In this new version, init always returns a non-null proxy.